flixel/graphics/atlas/AseAtlas.hx

```diff
@@ -0,0 +1,229 @@
+package flixel.graphics.atlas;
+
+import flixel.graphics.atlas.AtlasBase;
+
+typedef AseAtlasBase<T> = AtlasBase<T> &
+{
+	var meta:AseAtlasMeta;
+}
+
+/**
+ * Metadata attached to aseprite's atlas json, containing info about how the .aseprite file was
+ * set up.
+ */
+typedef AseAtlasMeta =
+{
+	/** The app used to save the file, often https://www.aseprite.org/ */
+	var app:String;
+	
+	/** The version of Aseprite used to export the json */
+	var version:String;
+	
+	/** Usually "I8" */
+	var format:String;
+	
+	/** The dimensions of the image file */
+	var size:AtlasSize;
+	
+	/** Usually "1" */
+	var scale:String;
+	
+	/**
+	 * The png file exported with this json, if one was
+	 * 
+	 * Note: Does not contain the relative filepath of that image (as of Aseprite 1.3-rc4)
+	 */
+	@:optional var image:String;
+	
+	/** A list of animation tags */
+	@:optional var frameTags:Array<AseAtlasTag>;
+	
+	/** A list of rects, 9-slices and pivots used in the Aseprite file, to be used however you desire */
+	@:optional var slices:Array<AseAtlasSlice>;
+	
+	/** A list of the layers used in the Aseprite file, to be used however you desire */
+	@:optional var layers:Array<AseAtlasLayer>;
+}
+
+typedef AseAtlasArray = AseAtlasBase<Array<AseAtlasFrame>>;
+
+typedef AseAtlasFrame = AtlasBase.AtlasFrame &
+{
+	duration:Int
+}
+
+typedef AseAtlas = AseAtlasBase<HashOrArray<AseAtlasFrame>>;
+
+typedef AseObject =
+{
+	/** The color used to display this object */
+	@:optional var color:AseAtlasColor;
+	
+	/** A message attached to this object */
+	@:optional var data:String;
+}
+
+/**
+ * Aseprite atlases use strings for colors (for some reason). This allows you to easily
+ * convert to a more usable format.
+ */
+abstract AseAtlasColor(String) to String
+{
+	/** Converts the underlying string to an actual color usable by flixel tools */
+	public function toFlxColor()
+	{
+		#if macro
+		// Avoid implicit imports that don't play nicely with macros on certain targets
+		return 0xFFFFFFFF;
+		#else
+		return flixel.util.FlxColor.fromString(this);
+		#end
```

Comment:
This is the only place we had an outward dependency, so I put this behind the !macro def.

Reply:
maybe we should do
```hx
// Note: FlxColor.fromString has imports that don't work with macros
return this == null ? null : Std.parseInt("0x" + this.substr(1));
```
so that this class can work in macros that use Aseprite files, and in the long term we can make FlxColor play nice with macros?